modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_detection.cc

```diff
@@ -71,42 +77,130 @@ bool PointPillarsDetection::Detect(const DetectionOptions& options,
   }
 
   // transform point cloud into an array
-  float* points_array = new float[original_cloud_->size() * 4];
-  PclToArray(original_cloud_, points_array, kNormalizingFactor);
+  float* points_array;
+  int num_points = original_cloud_->size();
+  int num_indexes;
+  std::vector<int> indexes;
```

Comment:
Use meaningful variable names for variable name "indexes". Please check it all over.

Reply:
Fixed.